internal/types/types.go

```diff
@@ -0,0 +1,38 @@
+// Package types defines all non-http types used in the STS.
+package types
+
+import (
+	"context"
+	"errors"
+)
+
+// UserInfo stores information about the user based on issuer/subject
+// pairs.
```

Comment:
Maybe include the relevant list of standard claims: https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims